File: Coding Ninjas/Easy/SetMatrixZeros.java
import java.util.*;

public class SetMatrixZeros {
    public static void SetZeros(int matrix[][]) {
        Set<Integer> rows = new HashSet<>();
        Set<Integer> cols = new HashSet<>();
        int m = matrix.length;
        int n = matrix[0].length;

        for ( int = 0; i < m; i++ ) {
            for ( int j = 0; j < n; j++ ) {
                if ( matrix[i][j] == 0 ) {
                    rows.add( i );
                    cols.add( j );
                }
            }
        }
        
        for ( int c : cols ) {
            for ( int i = 0; i < m; i++ ) {
                matrix[i][c] = 0;
            }
        }

        for ( int r : rows ) {
            Arrays.fill( matrix[r], 0 );
        }
    }
}

File: Coding Ninjas/Easy/Set Matrix Zeros.py
def setZeros( matrix: List[ List[int] ] ) -> None:
    n = len( matrix )
    m = len( matrix[0] )
    p = set()
    q = set()
    
    for i in range( n ):
        for j in range( m ):
            if matrix[i][j] == 0:
                p.add( i )
                q.add( j )
                
    for i in range( n ):
        if i in p:
            for j in range( m ):
                matrix[i][j] = 0
        else:
            for j in q:
                matrix[i][j] = 0